app/Actions/Album/Tracks.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Actions\Album;
+
+use App\Models\Album;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+
+class Tracks extends Action
+{
+	/**
+	 * @param string $albumID the ID of the album
+	 * @param mixed  $value   the value to be set
+	 */
+	public function set(string $albumID, UploadedFile $value): void
+	{
+		$album = $this->albumFactory->findAlbumOrFail($albumID, false);
+		if ($album->track_id != null) {
+			Storage::delete("tracks/$album->track_id.xml");
+		}
+
+		$new_track_id = uniqid();
+		Storage::putFileAs('tracks/', $value, "$new_track_id.xml");
+
+		if (Album::query()
+				->where('id', '=', $albumID)
+				->update(['track_id' => $new_track_id]) !== 1
+		) {
+			throw new ModelNotFoundException();
+		}
+	}
+
+	/**
+	 * Delete the track of an album.
+	 *
+	 * @param string $albumID the ID of the album
+	 */
+	public function delete(string $albumID): void
```

Comment:
I skip comments on this method. Basically, everything from above applies. ;-) And as I said, I would like to get rid of the entire action anyway.